extensions/wikia/ExactTargetUpdates/client/ExactTargetClient.class.php

```diff
@@ -0,0 +1,206 @@
+
+		return $this->sendRequest( self::CREATE_CALL, $oRequest );
+	}
+
+	public function deleteUserGroup( $userId, $group ) {
+		$request = ExactTargetRequestBuilder::getUserGroupDeleteBuilder()
+			->withUserId( $userId )
+			->withGroup( $group )
+			->build();
+
+		return $this->sendRequest( self::DELETE_CALL, $request );
+	}
+
+	public function createUserGroup( $userId, $group ) {
+		$request = ExactTargetRequestBuilder::getUserGroupCreateBuilder()
+			->withUserId( $userId )
+			->withGroup( $group )
+			->build();
+
+		return $this->sendRequest( self::CREATE_CALL, $request );
+	}
+
+	/**
+	 * Update or create user properties DataExtension object in ExactTarget by API request
+	 * that reflects Wikia user_properties table
+	 * @param int $userId User ID
+	 * @param array $userProperties key-value array ['property_name'=>'property_value']
+	 * @return bool
+	 * @throws \Exception
+	 */
+	public function updateUserProperties( $userId, array $userProperties ) {
+		$request = ExactTargetRequestBuilder::getPropertiesUpdateBuilder()
+			->withUserId( $userId )
+			->withProperties( $userProperties )
+			->build();
+
+		return $this->sendRequest( self::UPDATE_CALL, $request );
+	}
+
+	public function deleteUserProperties( $userId, $properties ) {
+		$request = ExactTargetRequestBuilder::getPropertiesDeleteBuilder()
+			->withUserId( $userId )
+			->withProperties( $properties )
+			->build();
+
+		return $this->sendRequest( self::DELETE_CALL, $request );
+	}
+
+	public function retrieveEmailByUserId( $userId ) {
+		$result = $this->retrieve( [ 'user_email' ], 'user_id', [ $userId ], ResourceEnum::USER );
+
+		return ( new UserEmailAdapter( $result ) )->getEmail();
+	}
+
+	public function retrieveUsersEdits( $usersIds ) {
+		$result = $this->retrieve(
+			[ 'user_id', 'wiki_id', 'contributions' ], 'user_id', $usersIds, ResourceEnum::USER_WIKI );
+
+		return ( new UserEditsAdapter( $result ) )->getEdits();
+	}
+
+	public function retrieveUserIdsByEmail( $email ) {
+		$result = $this->retrieve( [ 'user_id' ], 'user_email', [ $email ], ResourceEnum::USER );
+
+		return ( new UserIdsAdapter( $result ) )->getUsersIds();
+	}
+
+	public function updateUserEdits( $edits ) {
+		$request = ExactTargetRequestBuilder::getEditsUpdateBuilder()
+			->withEdits( $edits )
+			->build();
+
+		return $this->sendRequest( self::UPDATE_CALL, $request );
+	}
+
+	private function retrieve( array $properties, $filterProperty, array $filterValues, $resource ) {
+		$request = ExactTargetRequestBuilder::getRetrieveBuilder()
+			->withResource( $resource )
+			->withProperties( $properties )
+			->withFilterProperty( $filterProperty )
+			->withFilterValues( $filterValues )
+			->build();
+
+		return $this->sendRequest( self::RETRIEVE_CALL, $request );
+	}
+
+	protected function sendRequest( $type, $request ) {
+		// send first call
+		$response = $this->doCall( $type, $request, 0 );
+		if ( $response instanceof \stdClass && $response->OverallStatus === self::STATUS_OK ) {
```

Comment:
In this case we want to cover only cases we're sure and we expect, all others should throw errors so can be investigated in the future, hence the order.